### PR TITLE
fix(helm): add namespace to controller StatefulSet & DaemonSet

### PR DESCRIPTION
**Pull request by TECHNOFAB11**
## Pull Request template

Please, go through these steps before you submit a PR.

**Why is this PR required? What issue does it fix?**:
Tried installing zfs-localpv using Helm (helm template to be precise) but the controller StatefulSet (and node DaemonSet) could not be deployed because Kubernetes could not find the ServiceAccount in `default`. This adds the correct namespace to the StatefulSet & DaemonSet

**What this PR does?**:
See above

**Does this PR require any upgrade changes?**:
No

**If the changes in this PR are manually verified, list down the scenarios covered:**:
\-

**Any additional information for your reviewer?** :
\-

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

**Comment by sinhaashish** [comment]
I tried to replicate the issue , but could 
Steps 
1. **root@ashish-dev:~$ k get serviceaccount**
NAME                       SECRETS   AGE
default                    0         18h
2. **root@ashish-dev:~$ k get nodes**
NAME            STATUS   ROLES           AGE   VERSION
master-111320   Ready    control-plane   18h   v1.25.10
node-0-111320   Ready    <none>          18h   v1.25.10
node-1-111320   Ready    <none>          18h   v1.25.10
node-2-111320   Ready    <none>          18h   v1.25.10

3. root@ashish-dev:~$ **helm repo add openebs-zfslocalpv https://openebs.github.io/zfs-localpv**
"openebs-zfslocalpv" has been added to your repositories

4. **helm install zfs-localpv  openebs-zfslocalpv/zfs-localpv**
NAME: zfs-localpv
LAST DEPLOYED: Wed Aug 30 02:54:05 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
The OpenEBS ZFS LocalPV has been installed. Check its status by running:
$ kubectl get pods -n default -l role=openebs-zfs

For more information, visit our Slack at https://openebs.io/community or view
the documentation online at http://docs.openebs.io/.

5. **root@ashish-dev:~$ kubectl get pods -n default -l role=openebs-zfs**
NAME                       READY   STATUS    RESTARTS   AGE
zfs-localpv-controller-0   5/5     Running   0          8m30s
zfs-localpv-node-275rh     2/2     Running   0          8m30s
zfs-localpv-node-29vzw     2/2     Running   0          8m30s
zfs-localpv-node-9bvxt     2/2     Running   0          8m30s

`-n default ` is passed . to get the pods

**Comment by sinhaashish** [comment]
@TECHNOFAB11  can you run the install command with `--namespace <your-namespace>` and check ?

**root@ashish-dev:~/code/rust/mayastor-extensions(set-file)$ kubectl create ns ashish**
namespace/ashish created

**root@ashish-dev:~/code/rust/mayastor-extensions(set-file)$ helm install zfs-localpv openebs-zfslocalpv/zfs-localpv -n ashish**
NAME: zfs-localpv
LAST DEPLOYED: Wed Aug 30 06:06:58 2023
NAMESPACE: ashish
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
The OpenEBS ZFS LocalPV has been installed. Check its status by running:
$ kubectl get pods -n ashish -l role=openebs-zfs

For more information, visit our Slack at https://openebs.io/community or view
the documentation online at http://docs.openebs.io/.

**root@ashish-dev:~/code/rust/mayastor-extensions(set-file)$  kubectl get pods -n ashish -l role=openebs-zfs**
NAME                       READY   STATUS    RESTARTS   AGE
zfs-localpv-controller-0   5/5     Running   0          68s
zfs-localpv-node-f49bg     2/2     Running   0          68s
zfs-localpv-node-j482z     2/2     Running   0          68s
zfs-localpv-node-vwwv6     2/2     Running   0          68s


I see that you can install using `--namespace` .

**Comment by TECHNOFAB11** [comment]
@sinhaashish the issue is that `helm install` and `helm template` differ quite a bit in their functionality (there are a bunch of issues for these in the Helm repo but these are somehow "wontfix"). 
This here fixes it for `helm template` which does not set the namespace but only sets `.Release.namespace`.
Try using `helm template` and check for the DaemonSet and StatefulSet, they will not have any namespace.

TLDR; I cannot use `helm install` directly, so I had to change this for `helm template` to output roughly the same as `helm install`

**Review by sinhaashish — COMMENTED** [review]
@TECHNOFAB11  
I tried the below
1.  kubectl create ns ashish` 
2. `helm template zfs-localpv  openebs-zfslocalpv/zfs-localpv -n ashish > template1.yaml`
The statefulset doesn't have the namespace even after using your PR code. 
I dont think your PR fixes the issue if at all. 

Moreover its an issue with helm template , the `helm install` runs fine which is supported by us.

**Comment by TECHNOFAB11** [comment]
@sinhaashish for me it works, running `helm template deploy/helm/charts -n test | grep DaemonSet -C 10` shows the DaemonSet with the namespace "test".
The namespace variable used in this PR is literally already existant on all other resources and in most Helm charts everywhere, so I'm not sure why it's not showing up for you.
I'm also not sure what you mean with "even after using your PR code". You seem to be templating the current latest version which is not the same as this PRs version. Checkout this and try my command

**Review by sinhaashish — APPROVED** [review]
Tested & LGTM

**Review by hrudaya21 — APPROVED** [review]

**Comment by niladrih** [comment]
Closing this PR as these changes are included in #472. Thank you for your contribution @TECHNOFAB11!
